src/lib/NumberFormatting.js

```diff
@@ -85,6 +109,11 @@ class NumberFormatting { // eslint-disable-line no-unused-vars
             case 'cny':
                 return 'zh';
             default:
+                // Return the country from the currency code which is typically (but not necessarily) the first two
+                // letters (see https://en.wikipedia.org/wiki/ISO_4217#National_currencies), in the hope that it
+                // coincides with a locale.
+                // TODO oftentimes this results in the wrong locale, e.g. ARS (Argentinan Peso) -> AR (Arabic),
+                //  CAD (Canadian Dollar) -> CA (Catalan). Can we come up with a better heuristic?
```

Comment:
For these cases, they need to be added to the exemption list above, like for EUR and GBP.